lib/ansible/modules/database/postgresql/postgresql_facts.py

```diff
@@ -0,0 +1,921 @@
+'''
+
+RETURN = r'''
+version:
+  description: Database server version U(https://www.postgresql.org/support/versioning/).
+  returned: always
+  type: dict
+  sample: { "version": { "major": 10, "minor": 6 } }
+  contains:
+    major:
+      description: Major server version.
+      returned: always
+      type: int
+      sample: 11
+    minor:
+      description: Minor server version.
+      returned: always
+      type: int
+      sample: 1
+databases:
+  description: Information about databases.
+  returned: always
+  type: dict
+  sample:
+  - { "postgres": { "access_priv": "", "collate": "en_US.UTF-8",
+  "ctype": "en_US.UTF-8", "encoding": "UTF8", "owner": "postgres", "size": "7997 kB" } }
+  contains:
+    database_name:
+      description: Database name.
+      returned: always
+      type: dict
+      sample: template0
+      contains:
+        access_priv:
+          description: Database access privileges.
+          returned: always
+          type: str
+          sample: "=c/postgres_npostgres=CTc/postgres"
+        collate:
+          description:
+          - Database collation U(https://www.postgresql.org/docs/current/collation.html).
+          returned: always
+          type: str
+          sample: en_US.UTF-8
+        ctype:
+          description:
+          - Database LC_CTYPE U(https://www.postgresql.org/docs/current/multibyte.html).
+          returned: always
+          type: str
+          sample: en_US.UTF-8
+        encoding:
+          description:
+          - Database encoding U(https://www.postgresql.org/docs/current/multibyte.html).
+          returned: always
+          type: str
+          sample: UTF8
+        owner:
+          description:
+          - Database owner U(https://www.postgresql.org/docs/current/sql-createdatabase.html).
+          returned: always
+          type: str
+          sample: postgres
+        size:
+          description: Database size in bytes.
+          returned: always
+          type: str
+          sample: 8189415
+extensions:
+  description:
+  - Extensions U(https://www.postgresql.org/docs/current/sql-createextension.html).
+  returned: always
+  type: dict
+  sample:
+  - { "plpgsql": { "description": "PL/pgSQL procedural language",
+    "extversion": { "major": 1, "minor": 0 } } }
+  contains:
+    extdescription:
+      description: Extension description.
+      returned: if existent
+      type: str
+      sample: PL/pgSQL procedural language
+    extversion:
+      description: Extension description.
+      returned: always
+      type: dict
+      contains:
+        major:
+          description: Extension major version.
+          returned: always
+          type: int
+          sample: 1
+        minor:
+          description: Extension minor version.
+          returned: always
+          type: int
+          sample: 0
+    nspname:
+      description: Namecpase where the extension is.
+      returned: always
+      type: str
+      sample: pg_catalog
+languages:
```

Comment:
What I find confusing, is that we have two levels of info crossing here.
Basically, this info is only for the database you are connecting to.
But it is listed on the global level, next to the databases, so it looks as if it where global.

Reply:
I would propose same fix as the previous one (on extensions), but there is no pg_available_languages, so I would then skip this level entirely, or build it from info scanned from all databases, and adding a list of databases it was installed in.

Reply:
Very good notice! I'll think about it.